lib/tasks/attachments.rake

```diff
@@ -0,0 +1,72 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+namespace :attachments do
+  desc 'Copies all attachments from the current to the given storage.'
+  task :copy_to, [:to] => :environment do |task, args|
+    if args.empty?
+      puts 'rake attachments:copy_to[file|fog]'
+      exit 1
+    end
+
+    current_uploader = OpenProject::Configuration.file_uploader
+    target_uploader = OpenProject::Configuration.available_file_uploaders[args[:to].to_sym]
+
+    if target_uploader.nil?
+      puts "unknown storage: #{args[:to]}"
+      exit 1
+    end
+
+    if target_uploader == current_uploader
+      puts "attachments already in #{target_uploader} storage"
+      exit 1
+    end
+
+    TargetAttachment = Class.new Attachment
+    TargetAttachment.mount_uploader :file, target_uploader
+
+    def TargetAttachment.to_s
+      'attachment'
+    end
+
+    attachments = Attachment.all
+    attachments.each_with_index do |attachment, index|
+      print "Copying attachment #{attachment.id} [#{index + 1}/#{attachments.size}] \
+             (#{attachment.file.path}) to #{args[:to]} storage ... ".squish
+      STDOUT.flush
+
+      target = TargetAttachment.new
+      target.id = attachment.id
+      target.file = attachment.file.local_file
+      target.file.store!
```

Comment:
This could be placed into a class method of TargetAttachment. Just my personal preference, though.